app/Animal.php

```diff
@@ -2,23 +2,49 @@
 
 namespace App;
 
-use Illuminate\Support\Facades\DB;
 use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Facades\File;
 
 class Animal extends Model
 {
-    protected $fillable = ['name', 'image', 'about','category_id','shelter_id','birth_date'];
-
-//    public function categoryName()
-//    {
-//        $id = $this->category_id;
-//        return \App\AnimalCategory::where('id',$id)->value('title');
-//    }
-//    public function shelter()
-//    {
-//        $id = $this->shelter_id;
-//        return \App\Shelter::where('id',$id)->value('name_shelter');
-//    }
+    protected $fillable = ['name', 'about','category_id','shelter_id','birth_date'];
+
+    public static function add($values){
+        $animal = new static;
+        $animal->fill($values);
+        $animal->save();
+
+        return $animal;
+    }
+
+    public function uploadImage($image){
+        if ($image != null){
+
+            $imageName = time().$this->id.'.'.$image->extension();
+
+            $image->move('imageAnimals',$imageName);
+            $this->image = $imageName;
+            $this->save();
+        }
+    }
+
+    public function deleteImage(){
+        if ($this->image != null)
+            if (File::exists('imageAnimals/'.$this->image))
```

Comment:
1. File path should be const
2. image path should be uploads/images/animals/id/iamge_name